students/philip_behrend/lesson01/test_unit.py

```diff
@@ -0,0 +1,147 @@
+"""Unit testing for inventory management system"""
+
+from unittest import TestCase
+from unittest.mock import Mock
+from unittest.mock import patch
+from inventory_management.electric_appliance_class import ElectricAppliances
+from inventory_management.inventory_class import Inventory
+from inventory_management.furniture_class import Furniture
+import inventory_management.main as menu
+
+
+class InventoryTests(TestCase):
+    """Test for the Inventory class"""
+
+    def test_inv_init(self):
+        inv = Inventory(125, 'Test Item', 50, 80)
+        self.assertEqual(inv.product_code, 125)
+        self.assertEqual(inv.description, 'Test Item')
+        self.assertEqual(inv.market_price, 50)
+        self.assertEqual(inv.rental_price, 80)  
+
+    def test_return_dict(self):
+        inv = Inventory(125, 'Test Item', 50, 80)
+        inv_dict = inv.return_as_dictionary()
+        test_dict = {'product_code': 125,
+                    'description': 'Test Item',
+                    'market_price': 50,
+                    'rental_price': 80}
+        self.assertEqual(inv_dict, test_dict)
+
+class ElectricAppliancesTests(TestCase):
+    """ Test for ElectricApplicances class"""
+    
+    def test_elec_init(self):
+        elec = ElectricAppliances(54, 'Test Elec', 200, 300, 'John Deere', 2)
+        self.assertEqual(elec.brand, 'John Deere')
+        self.assertEqual(elec.voltage, 2)  
+
+    def test_return_dict(self):
+        elec = ElectricAppliances(54, 'Test Elec', 200, 300, 'John Deere', 2)
+        elec_dict = elec.return_as_dictionary()
+        test_dict = {'product_code': 54,
+                    'description': 'Test Elec',
+                    'market_price': 200,
+                    'rental_price': 300,
+                    'brand': 'John Deere',
+                    'voltage': 2}
+        self.assertEqual(elec_dict, test_dict)
+    
+class FurnitureTests(TestCase):
+    """ Test for Furniture class"""
+
+    def test_elec_init(self):
+        furn = Furniture(54, 'Test Elec', 200, 300, 'leather', 10)
+        self.assertEqual(furn.material, 'leather')
+        self.assertEqual(furn.size, 10)
+
+    def test_return_dict(self):
+        furn = Furniture(54, 'Test Elec', 200, 300, 'leather', 10)
+        furn_dict = furn.return_as_dictionary()
+        test_dict = {'product_code': 54,
+                    'description': 'Test Elec',
+                    'market_price': 200,
+                    'rental_price': 300,
+                    'material': 'leather',
+                    'size': 10}
+        self.assertEqual(furn_dict, test_dict)
+    
+class MainTests(TestCase):
+    """ Tests for functions in main """
+
+    def test_main_menu(self):
```

Comment:
Try changing this line to the following:

`def test_A_main_menu(self):`

Now, the question would be, why is this making it work? To get another clue, print the value being patched from _input()_.